frontend/src/sdk/useAppConfiguration.ts

```diff
@@ -1,18 +1,22 @@
 import * as React from 'react';
 import { commonFetch } from './commonFetch';
 import { getWSTokenSubProtocols } from './wsConfigs';
+import { createStore } from './createStore';
 import { AppInitSDK } from '@openshift/dynamic-plugin-sdk-utils';
-import { pluginStore } from 'Sdk/createStore';
-import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
+import { useScalprum } from '@scalprum/react-core';
+import { useChrome } from '@redhat-cloud-services/frontend-components/useChrome';
 
 type AppConfigurations = React.ComponentProps<typeof AppInitSDK>['configurations'];
 
 const useAppConfiguration = (): AppConfigurations | null => {
   const { auth } = useChrome();
   const [appConfigurations, setAppConfigurations] = React.useState<AppConfigurations | null>(null);
 
+  const { pluginStore } = useScalprum();
```

Comment:
If not running inside Chrome, the above `pluginStore` will be `undefined`, correct?

If so, we should probably ensure that HAC creates its own `PluginStore` if not provided via Chrome.

Reply:
Having a fallback store is fine with me. Got only have one question. Is it intended for hac-core to run somewhere else than chrome? If so, we would need more than just a fallback store. We would basically need multiple webpack entry points (product interface?) that would set the product context. For this case, there would be a "PluginStoreContext" that would be set in the product-specific entry point and the `useAppConfiguration` would consume the store from such context.

I think we might need a larger conversation about this. There was also a discussion/idea about "dissolving" hac-core and creating configurable plugins from it to act as the "product interface" instead.

What do you think about creating a fallback store that takes in the default webpack scope? For now. And then start designing a more generic solution that can handle multiple environments if necessary.

Reply:
Hac-core is specifically made as HCC app, I don't think it can run outside of chrome. There's also chrome hook https://github.com/openshift/hac-core/blob/main/frontend/src/App.tsx#L21 which pulls information from chrome (@Hyperkid123 it's somewhat similiar to what we are using here, isn't it?)

But I am all down for creating `PluginStore` if not provided. We have the https://github.com/openshift/dynamic-plugin-sdk/blob/main/packages/lib-utils/src/app/AppInitSDK.tsx#L61 I'd like to move that outside of sdk-utils and made it into a requirement (bootstrap a `PluginStore` and pass it into provider).

Reply:
:+1: on having a fallback `PluginStore` using the default webpack shared scope.

In future, it's possible to use different webpack shared scopes for different product contexts.

Reply:
@karelhala @Hyperkid123 Having a fallback `PluginStore` could also allow for developing and testing HAC Core pages outside Chrome, if that makes sense.

> We have the https://github.com/openshift/dynamic-plugin-sdk/blob/main/packages/lib-utils/src/app/AppInitSDK.tsx#L61 I'd like to move that outside of sdk-utils and made it into a requirement (bootstrap a `PluginStore` and pass it into provider).

Fully agreed :+1:  plugin SDK runtime init code typically depends on a specific product context so it makes perfect sense to move such code into the given application, instead of maintaining such code in an SDK utils package.

Reply:
OK, There is now a basic fallback plugin store.